weave/trace_server/refs_internal.py

```diff
@@ -90,20 +169,28 @@ def parse_internal_uri(uri: str) -> Union[InternalObjectRef, InternalTableRef]:
     if kind == "table":
         return InternalTableRef(project_id=project_id, digest=remaining[0])
     elif kind == "object":
-        name, version = remaining[0].split(":")
+        name_encoded, version = remaining[0].split(":")
+        name = urllib.parse.unquote_plus(name_encoded)
+        extra = remaining[1:]
+        if len(extra) == 1 and extra[0] == "":
+            extra = []
```

Comment:
While correct now, I feel these large blocks are prone to errors in future.  Can we pull the quote/unquote up one level?

Reply:
pulled into a common method